### DISCUSSION TITLE
Cannot solve a simple 3D transient model

### DISCUSSION THREAD

**Discussion by lcheng9** [su2code/SU2#2192] [Q&A]
Hi,

The attached project files are trying to solve a transient 3D cylinder model, the geometry is shown below. While the solver returns the error about the NaN value. Could you please guide me how to resolve this NaN value issue? Thank you.

![gui_model](https://github.com/su2code/SU2/assets/106984354/9dbbfe68-abcd-4ec9-926a-688021843a34)

```
Error in "void CSolver::SetResidual_RMS(const CGeometry*, const CConfig*)":
-------------------------------------------------------------------------
SU2 has diverged (NaN detected).
------------------------------ Error Exit -------------------------------
```

To run this model, please remove the ".txt" suffix after downloading these two files.
[mesh.su2.txt](https://github.com/su2code/SU2/files/13806689/mesh.su2.txt)
[su2.cfg.txt](https://github.com/su2code/SU2/files/13806690/su2.cfg.txt)

**Comment by rois1995** [comment]
Hi Icheng9, 

I think that the mesh is the problem. You are running a RANS case with an Euler mesh, thus without the boundary layer mesh. Try fixing this and the config accordingly.

**Reply by lcheng9** [reply]
Hi @rois1995 , Thank you so much for your response. 
I cannot find details of the boundary layer mesh for SU2, could you elaborate on the boundary layer mesh, such as difference with Euler mesh, file format, etc. I will be trying to create the boundary layer mesh for RANS.

**Reply by bigfooted** [reply]
You need to resolve gradients near the wall. Here is a video explaining inflation layers and wall meshing:
https://www.youtube.com/watch?v=1gSHN99I7L4

**Reply by lcheng9** [reply]
Hi @bigfooted, Thank you so much, the tutorial video is great. It looks the boundary layer meshing is an essential feature for a CFD mesher/pre-processor, I probably give it a try.
